qcodes/loops.py

```diff
@@ -249,22 +250,25 @@ def each(self, *actions):
         return ActiveLoop(self.sweep_values, self.delay, *actions,
                           then_actions=self.then_actions, station=self.station,
                           progress_interval=self.progress_interval,
-                          bg_task=self.bg_task, bg_min_delay=self.bg_min_delay)
+                          bg_task=self.bg_task, end_task=self.end_task, bg_min_delay=self.bg_min_delay)
 
-    def with_bg_task(self, task, min_delay=1):
+    def with_bg_task(self, task, end_task=None, min_delay=0.01):
         """
         Attaches a background task to this loop.
 
         Args:
             task: A callable object with no parameters. This object will be
                 invoked periodically during the measurement loop.
 
-            min_delay (default 1): The minimum number of seconds to wait
+            end_task: A callable object with no parameters. This object will be
```

Comment:
I think the doc string here should clarify that end_task is intended to clean up after or otherwise finish the background task work, rather than being for general cleanup.